packages/cf-wallet.js/src/app-instance.ts

```diff
@@ -32,17 +33,28 @@ export class AppInstance {
 
   readonly intermediaries?: Address[];
 
-  constructor(info: AppInstanceInfo, readonly provider: Provider) {
+  constructor(
+    info: AppInstanceInfo | AppInstanceJson,
+    readonly provider: Provider
+  ) {
     this.identityHash = info.identityHash;
-    this.appDefinition = info.appDefinition;
-    this.abiEncodings = info.abiEncodings;
-    this.myDeposit = info.myDeposit;
-    this.peerDeposit = info.peerDeposit;
-    this.timeout = info.timeout;
-    this.twoPartyOutcomeInterpreterParams =
-      info.twoPartyOutcomeInterpreterParams;
-    this.coinTransferInterpreterParams = info.coinTransferInterpreterParams;
-    this.intermediaries = info.intermediaries;
+
+    if (info["appInterface"] !== undefined) {
+      this.appDefinition = info["appInterface"].addr;
```

Comment:
Should we make an effort to make this a bit more type-safe? Something like:

```ts
if ("appInterface" in info) {
  // Replace T with the type that exposes `appInterface`.
  const { appInterface } = <T>info;
  // Assign data...
} else {
  // Replace T2 with the type that exposes all of these properties.
  const { appDefinition, abiEncodings, timeout } = <T2>info;
  // Assign data...
}
```

Reply:
Done